### ISSUE TITLE
Build instructions on Windows

### ISSUE THREAD

**Issue by northmaxdev** [fstl-app/fstl#123]
Apologies for the potentially redundant question, but are there any plans to include build instructions for Windows? Preferably via native toolchains, e.g. Visual Studio and MSVC.

**Comment by DeveloperPaul123** [comment]
That's doable, but Qt is the only dependency, so you can just do the standard `cmake` build

```cpp
cmake -S <fstl source folder> -B build -DQt5_DIR=<Path to Qt5/lib/cmake/Qt5>
cmake --build build
```

**Comment by DeveloperPaul123** [comment]
I see how the documentation is lacking though. Should be fixable. Thanks!

**Comment by Andryshik345** [comment]
@DeveloperPaul123 Can you tell what way do you use to compile windows version? I've been trying to do this in different ways for three days now and nothing has worked yet. Besides Qt and cmake, what compiler you need? Mingw? MSYS2? MSVC?

**Comment by DeveloperPaul123** [comment]
> [@DeveloperPaul123](https://github.com/DeveloperPaul123) Do you use cmake there? Is there any steps after `cmake --build build`? I also tried to use Github's CI without any luck. You can look here: [Andryshik345/fstl/actions](https://github.com/Andryshik345/fstl/actions)

@Andryshik345 Yes I do but I forgot to mention I don't use the Visual Studio generator. I use CMake + Ninja. This is the gist of the command I use: `cmake -S . -B build -DQt5_DIR=<Qt5 directory> -G "Ninja"`

**Comment by DeveloperPaul123** [comment]
> [@DeveloperPaul123](https://github.com/DeveloperPaul123) still fails on CI :( [Andryshik345/fstl/actions/runs/16004245100/job/45146820427](https://github.com/Andryshik345/fstl/actions/runs/16004245100/job/45146820427) Any thoughts? Maybe I forgot to install something with `choco`?

You've installed 2 versions of Qt5. Why not just use MSVC on Windows?

**Comment by Andryshik345** [comment]
> Why not just use MSVC on Windows?

@DeveloperPaul123 Sadly I just can't install Qt on my machine for some reasons (probably because I'm at Russia). And I suspect I'll get the same error anyway.

**Comment by DeveloperPaul123** [comment]
> > > > Why not just use MSVC on Windows?
> > > 
> > > 
> > > [@DeveloperPaul123](https://github.com/DeveloperPaul123) Sadly I just can't install Qt on my machine for some reasons (probably because I'm at Russia). And I suspect I'll get the same error anyway.
> > 
> > 
> > Are you setting up the CI so you can download a pre-built binary of `fstl` on Windows? I doubt you would get the same error if you didn't install 2 versions of Qt on the same CI and use MSVC instead of GNU as you're having linker errors and MSVC doesn't use `ld`.
> 
> I tried to configure cmake for MSVC before trying ninja and got this [Andryshik345/fstl/actions/runs/15978466537/job/45066939704](https://github.com/Andryshik345/fstl/actions/runs/15978466537/job/45066939704) Plus you keep saying me that I'm installing Qt twice, so it means CI already has installed Qt libs? Everyone on Internet install or build them, so I don't really know the correct way. If so, where they're installed on a VM then? Sorry if I turned this issue into a help thread.

You do need to install the libs. I thought it was installing twice because I saw a reference to C:\Tools\Qt\msvc2019 and the gnu build( see [here](https://github.com/Andryshik345/fstl/actions/runs/15978466537/job/45066939704#step:3:610) and [here](https://github.com/Andryshik345/fstl/actions/runs/15978466537/job/45066939704#step:5:5)), but after looking at the workflow file, I think it's just your path that is wrong since it seems like you're expecting the MSVC build of Qt but getting the MinGW one.

I'm not sure what the `qt5-default` choco package installs, but it seems to install the mingw build of Qt, which is probably not what you want. I would just use `aqt` directly instead or use something like this: https://github.com/jurplel/install-qt-action

**Comment by Andryshik345** [comment]
So, the whole story ended with the fact that Qt 5.15.x needs msvc2019 for building and Github Actions deprecated `windows-2019` runner on 2025-06-30 and the oldest they have now is `windows-2022`. Qt6 is too major hop and will probably fail to build here, so I guess I'll continue trying to build it locally.

**Comment by ristiisa** [comment]
> UPD: I finally did it. Thanks for help.

locally or in an action?
